luaui/Widgets/map_startbox.lua

```diff
@@ -489,52 +496,24 @@ function widget:Shutdown()
 	widgetHandler:DeregisterGlobal('GadgetCoopStartPoint')
 end
 
-local function DrawStartboxes3dWithStencil()
-	gl.DepthMask(false) --"BK OpenGL state resets", default is already false, could remove
-	if gl.DepthClamp then
-		gl.DepthClamp(true)
-	end
-
-	gl.DepthTest(true)
-	gl.StencilTest(true)
-	gl.ColorMask(false, false, false, false)
-	gl.Culling(false)
-
-	gl.StencilOp(GL.KEEP, GL.INVERT, GL.KEEP)
-
-	gl.CallList(startboxDListStencil)   --// draw
-
-	gl.Culling(GL.BACK)
-	gl.DepthTest(false)
-
-	gl.ColorMask(true, true, true, true)
-
-	gl.CallList(startboxDListColor)   --// draw
-
-	if gl.DepthClamp then
-		gl.DepthClamp(false)
-	end
-	gl.StencilTest(false)
-	gl.DepthTest(true)
-	gl.Culling(false)
-end
-
 --------------------------------------------------------------------------------
 --------------------------------------------------------------------------------
 
 function widget:DrawWorldPreUnit()
 	if autoReload then
 		startPolygonShader = LuaShader.CheckShaderUpdates(shaderSourceCache) or startPolygonShader
+		startConeShader = LuaShader.CheckShaderUpdates(coneShaderSourceCache) or startConeShader
 	end
 	DrawStartPolygons(false)
 end
 
+local cacheTable = {}
 function widget:DrawWorld()
 	gl.Blending(GL.SRC_ALPHA, GL.ONE_MINUS_SRC_ALPHA)
 
 	local time = Spring.DiffTimers(Spring.GetTimer(), startTimer)
 
-
+	clearInstanceTable(startConeVBOTable)
```

Comment:
Would be nice to skip recreating the whole vbo table here continuously but I guess it's not easy and probably not worth it.

Reply:
its not recreated, only its data is filled up, which is still cheaper than the old GL calls

Reply:
I have some benchmark that says so

Reply:
sure, I mean avoiding to clear instances and recreate, but yeah guess not very important really.